src/plays/age-calculator-by-react/useAge.jsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from 'react';
+
+const useAge = () => {
+  const [message, setMessage] = useState('');
+  const [year, setYear] = useState('-');
+  const [month, setMonth] = useState('-');
+  const [day, setDay] = useState('-');
+  const [birthday, setBirthday] = useState('');
+
+  const reset = (e) => {
+    e.preventDefault();
+    setMessage('Input your birthday');
+    setYear('-');
+    setMonth('-');
+    setDay('-');
+    setBirthday('');
+  };
+
+  const change = (e) => {
+    const { value } = e.target;
+    setBirthday(new Date(value));
+  };
+
+  const onSubmit = (e) => {
+    const months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
```

Comment:
Move the `months array` and `today's date` calculations outside the component to optimize performance.